sdk/java/logbat-sdk/src/main/java/info/logbat/config/LogBatConfigLoader.java

```diff
@@ -0,0 +1,158 @@
+package info.logbat.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Utility class for loading LogBat configuration from various file formats. Supports YAML (.yml,
+ * .yaml) and Properties (.properties) file formats. Configuration files are expected to be in the
+ * classpath.
+ *
+ * @author KyungMin Lee <a href="https://github.com/tidavid1">GitHub</a>
+ * @version 0.1.2
+ * @since 0.1.2
+ */
+@SuppressWarnings("unchecked")
+public class LogBatConfigLoader {
+
+    private static final Map<String, String> CONFIG_MAP = new HashMap<>();
+    private static final List<ConfigLoader> LOADERS = List.of(
+        new YamlConfigLoader("application.yml"),
+        new YamlConfigLoader("application.yaml"),
+        new PropertiesConfigLoader("application.properties")
+    );
+
+    /**
+     * Private constructor to prevent instantiation of this utility class.
+     *
+     * @throws UnsupportedOperationException if an attempt is made to instantiate this class
+     */
+    private LogBatConfigLoader() {
+        throw new UnsupportedOperationException("Utility class");
+    }
+
+    /**
+     * Loads the LogBat configuration from the first available configuration file. Tries to load
+     * from YAML files first, then falls back to Properties file.
+     *
+     * @return {@link LogBatConfig} object containing the loaded configuration
+     */
+    public static LogBatConfig loadConfig() {
+        for (ConfigLoader loader : LOADERS) {
+            if (loader.load(loader.filename)) {
+                break;
+            }
+        }
+        return new LogBatConfig(CONFIG_MAP);
+    }
+
+    /**
+     * Abstract base class for configuration loaders.
+     */
+    private abstract static class ConfigLoader {
+
+        protected String filename;
```

Comment:
이렇게 final 키워드로 `filename` 이 있음을 보장하면 어떨까요?
```suggestion
        protected final String filename;

        protected ConfigLoader(String filename) {
            this.filename = filename;
        }
```